frontend/packages/console-shared/src/types/resource.ts

```diff
@@ -35,6 +35,8 @@ export type OverviewItem = {
   configurations?: K8sResourceKind[];
   ksservices?: K8sResourceKind[];
   revisions?: K8sResourceKind[];
+  pipeline?: K8sResourceKind;
+  pipelineRuns?: K8sResourceKind[];
```

Comment:
Can we avoid having these listed here?
I see the knative snuck in some values here. We'd need to clean those up as well.
Knative and pipelines shouldn't bleed into the console-shared.
Where are these being used?

Reply:
fixed in https://github.com/openshift/console/pull/3014/commits/65c3b41d3e36e8886063507b03df00ba95c78ab8

Reply:
@christianvogt @sahil143 
So now that these are out of the types here - it's confusing reading the code. 

```
const createTopologyNodeData = (dc: OverviewItem, cheURL?: string): TopologyDataObject => {
```
I read this, and go "okay, `dc` is of type `OverviewItem`, let's see what comes out of that type" - and don't find pipeline or pipelineRuns. However, if I now start up the code and throw a debugger/log in, oh look, there it is.

I get the purity of the package, but we need to consider how to address this kind of stuff across packages. TypeScript is here to help us, we definitely should be showcasing all relevant properties of an object, especially when it's typed.